src/librustc_trans/base.rs

```diff
@@ -2734,6 +2730,33 @@ pub fn trans_crate<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
         };
     }
 
+    // Instantiate translation items without filling out definitions yet...
```

Comment:
Nit: I'd prefer to see some methods here, like

```rust
predefine_trans_items(&crate_context_list);
define_trans_items(&crate_context_list);
```

rather than having these methods bodies inline, particularly as they are going to get more complex as the match arms grow.